examples/2d/rotation.rs

```diff
@@ -0,0 +1,253 @@
+            rotation_speed: f32::to_radians(90.0), // degrees per second
+        });
+}
+
+/// Demonstrates applying rotation and movement based on keyboard input.
+fn player_movement_system(
+    keyboard_input: Res<Input<KeyCode>>,
+    mut query: Query<(&Player, &mut Transform)>,
+) {
+    let (ship, mut transform) = query.single_mut();
+
+    let mut rotation_factor = 0.0;
+    let mut movement_factor = 0.0;
+
+    if keyboard_input.pressed(KeyCode::Left) {
+        rotation_factor += 1.0;
+    }
+
+    if keyboard_input.pressed(KeyCode::Right) {
+        rotation_factor -= 1.0;
+    }
+
+    if keyboard_input.pressed(KeyCode::Up) {
+        movement_factor += 1.0;
+    }
+
+    // create the change in rotation around the Z axis (pointing through the 2d plane of the screen)
+    let rotation_delta = Quat::from_rotation_z(rotation_factor * ship.rotation_speed * TIME_STEP);
+    // update the ship rotation with our rotation delta
+    transform.rotation *= rotation_delta;
+
+    // get the ship's forward vector by applying the current rotation to the ships initial facing vector
+    let movement_direction = transform.rotation * Vec3::Y;
+    // get the distance the ship will move based on direction, the ship's movement speed and delta time
+    let movement_distance = movement_factor * ship.movement_speed * TIME_STEP;
+    // create the change in translation using the new movement direction and distance
+    let translation_delta = movement_direction * movement_distance;
+    // update the ship translation with our new translation delta
+    transform.translation += translation_delta;
+
+    // bound the ship within the invisible level bounds
+    let extents = Vec3::from((BOUNDS / 2.0, 0.0));
+    transform.translation = transform.translation.min(extents).max(-extents);
+}
+
+/// Demonstrates snapping the enemy ship to face the player ship immediately.
+fn snap_to_player_system(
+    mut query: Query<&mut Transform, (With<SnapToPlayer>, Without<Player>)>,
+    player_query: Query<&Transform, With<Player>>,
+) {
+    let player_transform = player_query.single();
+    // get the player translation in 2d
+    let player_translation = player_transform.translation.xy();
+
+    for mut enemy_transform in query.iter_mut() {
+        // get the enemy ship forward vector in 2d (already unit length)
+        let enemy_forward = (enemy_transform.rotation * Vec3::Y).xy();
+        // get the vector from the enemy ship to the player ship in 2d and normalize it.
+        let to_player = (player_translation - enemy_transform.translation.xy()).normalize();
+
+        // get the quaternion to rotate from the current enemy facing direction to the direction
+        // facing the player
+        let rotate_to_player = Quat::from_rotation_arc(
+            Vec3::from((enemy_forward, 0.0)),
+            Vec3::from((to_player, 0.0)),
+        );
+
+        // rotate the enemy to face the player
+        enemy_transform.rotation *= rotate_to_player;
```

Comment:
Generally when performing rotations like this I avoid using angles as that typically requires using expensive trig operations like `acos`. It is usually unnecessary to use trigonometry when most things can be achieved with linear algebra, which is what I wanted to demonstrate here.

Reply:
I was not talking about angles. Can't you calculate the absolute rotation with linear algebra, and assign it as the current rotation, instead of calculating the relative one and adding it to the current one?

Reply:
Ah right, I think I understand. Possibly using the rotation arc method between the initial facing direction (`Vec3::Y`) and the `to_player` ship vector would do that.

Reply:
Yes exactly

Reply:
Done